### PR TITLE
Update HUA analysis visualization

### PR DESCRIPTION
**Pull request by ywkim312**
There are several changes in this PR
1. There were updates elements for the visualization, such as sum data by count for person id or percent option
2. Renaming the class. Originally, it was named as IncoreHousingUnitAllocation, but according to Nathanael, this should be used for other analysis in the future cases. Originally, analysis visualization has classes with the same name as the dataType, so the user can fire the visualization by just using `viz.visualize(dataset)`. Then, the visualize method automatically imports the corresponding class that matches to dataType. Therefore, I added very simple mapping to dataType to this class in the code. If there are more analysis that needs this visualization, by simply adding the dataType to given list, it will work as it should be. Otherwise, the visualize will import the class with the dataType. 

You can test the code using  this code, or use pyincore-analysis-viz-example.ipynb in incore-docs.

```from pyincore import IncoreClient
from pyincore.analyses.housingunitallocation.housingunitallocation import HousingUnitAllocation
from pyincore_viz import AnalysisViz


def test_hua_visualization(client):
    # Joplin
    building_inv = "5f218e36114b783cb0b01833"
    housing_unit_inv = "5df7ce61425e0b00092d0013"  # 2ev3
    address_point_inv = "5df7ce0d425e0b00092cffee"  # 2ev2
    result_name = "IN-CORE_2ev3_HUA"

    seed = 1238
    iterations = 2

    spa = HousingUnitAllocation(client)
    spa.load_remote_input_dataset("buildings", building_inv)
    spa.load_remote_input_dataset("housing_unit_inventory", housing_unit_inv)
    spa.load_remote_input_dataset("address_point_inventory", address_point_inv)

    spa.set_parameter("result_name", result_name)
    spa.set_parameter("seed", seed)
    spa.set_parameter("iterations", iterations)

    spa.run_analysis()

    # test viz
    who = "Total Population"
    what = "by Race, Ethniciy, and Tenure Status"
    when = "2010"
    where = "Robeson County NC"
    row_index = 'Race Ethnicity'
    col_index = 'Tenure Status'
    kwargs = {"who": who, "what": what, "when": when, "where": where, "row_index": row_index, "col_index": col_index}

    dataset = spa.get_output_dataset("result")
    df = AnalysisViz.visualize(dataset, **kwargs)

    print(df.data)

    assert df.index[0] == '1 White alone, Not Hispanic'


if __name__ == "__main__":
    INCORE_DEV_API_URL = "https://incore-dev.ncsa.illinois.edu"
    client = IncoreClient(INCORE_DEV_API_URL)

    test_hua_visualization(client)
```

### CODE REVIEW

**Comment by navarroc** [comment]
Can you fix the merge conflict?

**Comment by ywkim312** [comment]
> Can you fix the merge conflict?

fixed

File: pyincore_viz/analysisviz.py [hunk 1]
```diff
@@ -19,6 +19,9 @@ def visualize(dataset, **kwargs):
                 None
 
         """
+        # data types that needs to use pop_results_table visualization
+        pop_result_table_data_types = ['incorehousingunitallocation']
```
Comment (navarroc) [inline pyincore_viz/analysisviz.py:23]: We may need to think about how we can maybe make this more dynamic, some way to "register" additional types without modifying the code itself.

Comment (navarroc) [reply pyincore_viz/analysisviz.py:23]: Not for now, but as an improvement we may want to see if we can come up with a more dynamic way to do this.

Comment (ywkim312) [reply pyincore_viz/analysisviz.py:23]: that is exactly what I was thinking but I was not able to find any better solution. Do you have any suggestion?

Comment (ywkim312) [reply pyincore_viz/analysisviz.py:23]: we have very similar situation in incore service for registering hazard data type but it is using the same way. Just put datatypee list in the code

File: pyincore_viz/analysis/popresultstable.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    """
+
+    @staticmethod
+    def visualize(dataset, **kwargs):
+        """visualize Population related dataframe.
+        Args:
+            dataset (obj): Housing unit Inventory,
+                Person Record, or Job Inventory dataset object.
+            kwargs (kwargs): Keyword arguments for visualization title.
+            who (str): Who does the data represent (unit of analysis):
```
[the remark this answers is not in the document]
Comment (ywkim312) [reply pyincore_viz/analysis/popresultstable.py:26]: This has been done during the last release by Chen and Rob, Jong wanted to make the way to use just .visualize() without any other inputs and this kwargs was probably the solution that they found with hours of testing. I believe that there are some other methods that are using kwargs and the documenting it is probably the same as this one. I don't mind if we  discuss about this to make it to the other way, and I would like to you to make a verdict.

[the remark this answers is not in the document]
Comment (navarroc) [reply pyincore_viz/analysis/popresultstable.py:26]: Fair enough, was just curious. I think this change looks fine, approving.

**Pull request merged**